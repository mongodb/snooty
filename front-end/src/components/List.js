import React, { Component } from 'react';
import ComponentFactory from '../components/ComponentFactory';

export default class List extends Component {

  render() {
    return (
      <ul>
        {
          this.props.nodeData.children.map((item, index) => {
            return (
              <li key={ index }>
                {
                  item.children.map((listItem, listItemIndex) => {
                    return <ComponentFactory { ...this.props } nodeData={ listItem } key={ listItemIndex } />
                  })
                }
              </li>
            )
          })
        }
      </ul>
    )
  }

}

