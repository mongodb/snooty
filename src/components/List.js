import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from './ComponentFactory';

const List = props => {
  const { nodeData } = props;
  return (
    <ul>
      {nodeData.children.map((item, index) => (
        <li key={index}>
          {item.children.map((listItem, listItemIndex) => (
            <ComponentFactory {...props} nodeData={listItem} key={listItemIndex} />
          ))}
        </li>
      ))}
    </ul>
  );
};

List.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
  }).isRequired,
};

export default List;
