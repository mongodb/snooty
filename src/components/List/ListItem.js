import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from '../ComponentFactory';

const ListItem = ({ nodeData, ...rest }) => (
  <li>
    {nodeData.children.map((child, index) => (
      <ComponentFactory
        {...rest}
        nodeData={child}
        key={index}
        // Include <p> tags in <li> if there is more than one paragraph
        skipPTag={nodeData.children.length === 1}
      />
    ))}
  </li>
);

ListItem.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
  }).isRequired,
};

export default ListItem;
