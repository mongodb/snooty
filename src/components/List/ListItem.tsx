import React from 'react';
import { cx, css } from '@leafygreen-ui/emotion';
import ComponentFactory from '../ComponentFactory';
import { ParentNode } from '../../types/ast';

const listParagraphStyles = css`
  ::marker {
    color: var(--font-color-primary);
  }
  & > p {
    margin-bottom: 8px;
  }
`;

export type ListItemProps = {
  nodeData: ParentNode;
};

const ListItem = ({ nodeData, ...rest }: ListItemProps) => (
  <li className={cx(listParagraphStyles)}>
    {nodeData.children.map((child, index) => (
      <ComponentFactory {...rest} nodeData={child} key={index} skipPTag={false} />
    ))}
  </li>
);

export default ListItem;
