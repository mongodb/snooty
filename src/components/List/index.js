import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from '../ComponentFactory';

const enumtypeMap = {
  arabic: '1',
  loweralpha: 'a',
  upperalpha: 'A',
  lowerroman: 'i',
  upperroman: 'I',
};

const List = ({ nodeData: { children, enumtype, startat }, ...rest }) => {
  const ListTag = enumtype === 'unordered' ? 'ul' : 'ol';
  const attributes = {};
  if (enumtype in enumtypeMap) {
    attributes.type = enumtypeMap[enumtype];
  }
  if (startat) {
    attributes.start = startat;
  }
  return (
    <ListTag {...attributes}>
      {children.map((listChild, index) => (
        <ComponentFactory {...rest} nodeData={listChild} key={index} />
      ))}
    </ListTag>
  );
};

List.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
    enumtype: PropTypes.string.isRequired,
    startat: PropTypes.number,
  }).isRequired,
};

export default List;
