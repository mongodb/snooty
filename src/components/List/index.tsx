import React from 'react';
import { css, cx } from '@leafygreen-ui/emotion';
import ComponentFactory from '../ComponentFactory';
import { ListNode } from '../../types/ast';

const enumtypeMap = {
  arabic: '1',
  loweralpha: 'a',
  upperalpha: 'A',
  lowerroman: 'i',
  upperroman: 'I',
};

type EnumTypeKey = keyof typeof enumtypeMap;

const listStyles = css`
  margin-top: 0px;
`;

export type ListProps = {
  nodeData: ListNode;
};

const List = ({ nodeData: { children, enumtype, startat }, ...rest }: ListProps) => {
  const ListTag = enumtype === 'unordered' ? 'ul' : 'ol';

  const typeAttr = enumtype && enumtype in enumtypeMap ? enumtypeMap[enumtype as EnumTypeKey] : undefined;
  const startAttr = startat ?? undefined;

  return (
    <ListTag
      className={cx(listStyles)}
      {...(typeAttr ? { type: typeAttr as '1' | 'a' | 'A' | 'i' | 'I' } : {})}
      {...(startAttr !== undefined ? { start: startAttr } : {})}
    >
      {children.map((listChild, index) => (
        <ComponentFactory {...rest} nodeData={listChild} key={index} />
      ))}
    </ListTag>
  );
};

export default List;
