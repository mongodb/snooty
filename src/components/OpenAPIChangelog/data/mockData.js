export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
