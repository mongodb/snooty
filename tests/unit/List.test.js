import React from 'react';
import { render } from 'enzyme';
import List from '../../src/components/List';

// data for this component
import mockData from './data/List.test.json';

it('List renders correctly', () => {
  const tree = render(<List nodeData={mockData} />);
  expect(tree).toMatchSnapshot();
});
