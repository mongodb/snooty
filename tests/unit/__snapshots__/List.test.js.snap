// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <li
    key="0"
  >
    <ComponentFactory
      key="0"
      nodeData={
        Object {
          "children": Array [
            Object {
              "position": Object {
                "start": Object {
                  "line": 16,
                },
              },
              "type": "text",
              "value": "create a free account on Atlas, MongoDB's managed database solution in the cloud",
            },
          ],
          "position": Object {
            "start": Object {
              "line": 16,
            },
          },
          "type": "paragraph",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <ComponentFactory
      key="0"
      nodeData={
        Object {
          "children": Array [
            Object {
              "position": Object {
                "start": Object {
                  "line": 17,
                },
              },
              "type": "text",
              "value": "create a free MongoDB ",
            },
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "start": Object {
                      "line": 17,
                    },
                  },
                  "type": "text",
                  "value": "cluster",
                },
              ],
              "position": Object {
                "start": Object {
                  "line": 17,
                },
              },
              "type": "emphasis",
            },
            Object {
              "position": Object {
                "start": Object {
                  "line": 17,
                },
              },
              "type": "text",
              "value": ", or group of database instances.",
            },
          ],
          "position": Object {
            "start": Object {
              "line": 17,
            },
          },
          "type": "paragraph",
        }
      }
    />
  </li>
</ul>
`;
